wp-includes/class-wp-service-workers.php

```diff
@@ -152,7 +152,7 @@ public function serve_request() {
 		$file_hash = md5( $output );
 		@header( "ETag: $file_hash" ); // phpcs:ignore Generic.PHP.NoSilencedErrors.Discouraged, WordPress.PHP.NoSilencedErrors.Discouraged
 
-		$etag_header = isset( $_SERVER['HTTP_IF_NONE_MATCH'] ) ? trim( $_SERVER['HTTP_IF_NONE_MATCH'] ) : false;
+		$etag_header = isset( $_SERVER['HTTP_IF_NONE_MATCH'] ) ? trim( sanitize_text_field( $_SERVER['HTTP_IF_NONE_MATCH'] ) ) : false;
 		if ( $file_hash === $etag_header ) {
```

Comment:
```suggestion
		if ( $if_none_match === $etag ) {
```

Reply:
For more on what should be done here, refer to what core is doing: https://github.com/WordPress/wordpress-develop/blob/803eab428d54755acf83403a49bb1cb9c3663c13/src/wp-includes/class-wp.php#L456-L476